swag/client/client.py

```diff
@@ -74,117 +85,182 @@ async def process(self, message):
         try:
             if message.content.startswith("!$wagdmin"):
                 await execute_swagdmin_command(self, message)
-            elif message.content.startswith("!$wag"):
-                await execute_swag_command(self, message)
-            elif message.content.startswith("!$tyle"):
-                await execute_style_command(self, message)
-            elif message.content.startswith("!€agnotte"):
-                await execute_cagnotte_command(self, message)
-#            elif message.content.startswith("!Yfu"):
-#                await execute_yfu_command(self, message)
-        except NotEnoughSwagInBalance:
+        except InvalidTimeZone as e:
             await message.channel.send(
-                f"{message.author.mention} ! Tu ne possèdes pas assez de $wag pour "
-                "faire cette transaction, vérifie ton solde avec `!$wag solde`"
+                f"{e.name}, n'est pas un nom de timezone valide !\n"
+                "Vérifie le nom correct sur "
+                "https://en.wikipedia.org/wiki/List_of_tz_database_time_zones, "
+                "à la colone `TZ database name`."
             )
-        except InvalidSwagValue:
+        except TimeZoneFieldLocked as e:
             await message.channel.send(
-                f"{message.author.mention}, la valeur que tu as écrite est "
+                "Tu viens déjà de changer de timezone. Tu ne pourras effectuer "
+                f"à nouveau cette opération qu'après le {e.date}. Cette mesure "
+                "vise à empécher l'abus de minage, merci de ta compréhension.\n\n"
+                "*L'abus de minage est dangereux pour la santé. À Miner avec "
+                "modération. Ceci était un message de la Fédération Bobbyique du "
+                "Minage*"
+            )
+
+
+##TODO : Possible amélioration
+## Créer un ClientCommand qui sera un Cog qui importera les autres Cog (Swag, Style, Yfu, Swagadmin...)
+
+
+class ClientError(commands.Cog):
+    """
+    Cog qui écoute les erreurs envoyées par les slash commands et envoie le bon message d'erreur associé
+    """
+
+    ##TODO check the source
+    @commands.Cog.listener()
+    async def on_slash_command_error(
+        self,
+        interaction: disnake.ApplicationCommandInteraction,
+        error: commands.CommandInvokeError,
+    ):
+
+        if type(error.original) == AccountAlreadyExist:
```

Comment:
This is super ugly

Reply:
rofl

Reply:
Maybe you got a suggestion ?

Reply:
Unfortunately no
I was more whining on our fate than commenting the design
One real bonafide question though, how does this interacts with the error logging, when the error is unknow?

Reply:
I try to handle that with the last commit. It display the traceback in the console.

However the problem is it do not send a error message to the user like before with the message "Une erreur est survenue les développeurs sont prévenus" Maybe we should just move this message just after the traceback.